plugins/python/redisgears_python.c

```diff
@@ -1751,6 +1759,509 @@ static PyObject* atomicExit(PyObject *self, PyObject *args){
+    return _loadTensorToDAG(self, args, RedisAI_DAGAddTensorSet);
+}
+
+static PyObject* DAGAddTensorGet(PyObject *self, PyObject *args) {
+    verifyRedisAILoaded();
+    PyDAGRunner *pyDag = (PyDAGRunner *)self;
+    if (!_IsValidDag(pyDag, RedisAI_DAGAddTensorGet)) {
+        return NULL;
+    }
+
+    if(PyTuple_Size(args) != 1){
+        PyErr_SetString(GearsError, "Wrong number of args to TensorSet op");
+        return NULL;
+    }
+    PyObject* tensorName = PyTuple_GetItem(args, 0);
+    if(!PyUnicode_Check(tensorName)){
+        PyErr_SetString(GearsError, "Tensor name argument must be a string");
+        return NULL;
+    }
+
+    const char* inputNameStr = PyUnicode_AsUTF8AndSize(tensorName, NULL);
+    RedisAI_DAGAddTensorGet(pyDag->dag, inputNameStr);
+    Py_INCREF(self);
+    return self;
+}
+
+static RAI_DAGRunOp *_createModelRunOp(const char *modelNameStr, RAI_Error *err) {
+
+    RedisGears_LockHanlderAcquire(staticCtx);
+    RedisModuleString* keyRedisStr = RedisModule_CreateString(staticCtx, modelNameStr, strlen(modelNameStr));
+    RAI_Model *model;
+    if(RedisAI_GetModelFromKeyspace(staticCtx, keyRedisStr, &model, REDISMODULE_READ, err) != REDISMODULE_OK) {
+        RedisModule_FreeString(staticCtx, keyRedisStr);
+        RedisGears_LockHanlderRelease(staticCtx);
+        RedisModule_FreeThreadSafeContext(staticCtx);
+        PyErr_SetString(GearsError, RedisAI_GetError(err));
+        return NULL;
+    }
+
+    RedisModule_FreeString(staticCtx, keyRedisStr);
+    RAI_DAGRunOp *modelRunOp = RedisAI_DAGCreateModelRunOp(model);
+    RedisGears_LockHanlderRelease(staticCtx);
+    return modelRunOp;
+}
+
+static PyObject* DAGAddModelRun(PyObject *self, PyObject *args, PyObject *kargs) {
+    verifyRedisAILoaded();
+    PyDAGRunner *pyDag = (PyDAGRunner *)self;
+    if (!_IsValidDag(pyDag, RedisAI_DAGCreateModelRunOp)) {
+        return NULL;
+    }
+
+    PyObject* modelName = GearsPyDict_GetItemString(kargs, "name");
+    if (!modelName) {
+        PyErr_SetString(GearsError, "Model key name was not given");
+        return NULL;
+    }
+    if(!PyUnicode_Check(modelName)){
+        PyErr_SetString(GearsError, "Model name argument must be a string");
+        return NULL;
+    }
+    RAI_Error *err;
+    RedisAI_InitError(&err);
+    const char* modelNameStr = PyUnicode_AsUTF8AndSize(modelName, NULL);
+
+    // Create MODELRUN op after bringing model from keyspace (raise an exception if it does not exist)
+    RAI_DAGRunOp *modelRunOp = _createModelRunOp(modelNameStr, err);
+    if (modelRunOp == NULL) {
+        RedisAI_FreeError(err);
+        return NULL;
+    }
+
+    // Add to the modelRun op with its inputs and output keys and insert it to the DAG
+    PyObject* modelInputs = GearsPyDict_GetItemString(kargs, "inputs");
+    if (!modelInputs) {
+        PyErr_SetString(GearsError, "Must specify model inputs");
+        goto cleanup;
+    }
+    PyObject* inputsIter = PyObject_GetIter(modelInputs);
```

Comment:
I think you need to free the iterator after you are done using it.. please check.